pallets/pallet-did-lookup/src/tests.rs

```diff
@@ -554,3 +557,80 @@ fn test_update_deposit_unauthorized() {
 			);
 		})
 }
+
+// #############################################################################
+// migrate
+
+#[test]
+fn partial_migration() {
+	let deposit_account = || generate_acc32(usize::MAX);
+
+	ExtBuilder::default()
+		.with_balances(vec![(
+			deposit_account(),
+			<Test as crate::Config>::Deposit::get() * 50_000,
+		)])
+		.build()
+		.execute_with(|| {
+			for i in 0..50 {
+				add_legacy_association::<Test>(
+					deposit_account(),
+					generate_did(i),
+					generate_acc32(i),
+					<Test as crate::Config>::Deposit::get(),
+				);
+			}
+			assert_eq!(
+				get_mixed_storage_iterator::<Test>(None).fold((0usize, 0usize, 0usize), |acc, key| match key {
+					MixedStorageKey::V1(_) => (acc.0 + 1, acc.1, acc.2),
+					MixedStorageKey::V2(LinkableAccountId::AccountId20(_)) => (acc.0, acc.1 + 1, acc.2),
+					MixedStorageKey::V2(LinkableAccountId::AccountId32(_)) => (acc.0, acc.1, acc.2 + 1),
+				}),
+				(50usize, 0usize, 0usize),
+				"We should only have V1 keys"
+			);
+			assert_eq!(MigrationStateStore::<Test>::get(), MigrationState::PreUpgrade);
+
+			// The tuple contains the number of keys in storage
+			// (old Key, 20Bytes Account key, 32Bytes Account Key)
+			let expected_key_distributions = [
```

Comment:
I expect this test to be removed before the next polkadot upgrade. Also the storage cannot change, otherwise we would have issues decoded/finding the right keys on our production chains

Reply:
I was not referring to the storage itself, but to the order in which the iterator would yield the different keys. For example, if tomorrow the iterator yields keys in ascending order instead of descending, the number of keys that can be decoded could change. But I get it that we will remove the tests soon enough anyway.